impress-org/givewp
# Feature: Preview API

## Description

This PR introduces a new Preview API that is used to pass the state changes in design mode to the form inside the iframe. Preview API is a simple pub/sub system that hooks up in the form state which makes all changes in design mode show up instantly on the screen without reloading the form.

## Affects

Form Builder
Donation form

## Visuals
![Screen Capture_select-area_20231023123856](https://github.com/impress-org/givewp/assets/4222590/ec98dea8-5540-46da-adf1-fee2a7dace15)


## Testing Instructions

1. Create a new form
2. Enter design mode
3. Try every  form customization option in the right-hand side panel 
4. Verify that form settings are saved correctly by checking the form on the frontend



## Pre-review Checklist





Plugin zip for QA: [https://github.com/impress-org/givewp/suites/17790212597/artifacts/1019602715](https://github.com/impress-org/givewp/suites/17790212597/artifacts/1019602715)

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205783384593885
  - https://app.asana.com/0/0/1205269709893095
